src/v/utils/external_process.cc

```diff
@@ -0,0 +1,162 @@
+ * by the Apache License, Version 2.0
+ */
+
+#include "external_process.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/timed_out_error.hh>
+#include <seastar/core/when_all.hh>
+#include <seastar/core/with_timeout.hh>
+#include <seastar/util/later.hh>
+#include <seastar/util/process.hh>
+
+#include <optional>
+
+namespace external_process {
+ss::future<external_process> external_process::create_external_process(
+  std::vector<ss::sstring> command,
+  std::optional<std::vector<ss::sstring>> env) {
+    vassert(
+      !command.empty(),
+      "Should not be passed an empty command list to "
+      "create_external_process");
+
+    auto path = std::filesystem::path(command[0]);
+    ss::experimental::spawn_parameters params{
+      .argv = std::move(command),
+      .env = env.has_value() ? std::move(env).value()
+                             : std::vector<ss::sstring>{}};
+
+    auto p = co_await ss::experimental::spawn_process(path, params);
+
+    co_return external_process(
+      std::move(p), std::move(path), std::move(params));
+}
+
+external_process::external_process(
+  ss::experimental::process p,
+  std::filesystem::path path,
+  ss::experimental::spawn_parameters params)
+  : _process(std::move(p))
+  , _path(std::move(path))
+  , _params(std::move(params))
+  , _terminate_mutex("external_process::terminate")
+  , _wait_mutex("external_process::wait") {}
+
+external_process::external_process(external_process&& rhs) noexcept
+  : _process(std::move(rhs._process))
+  , _path(std::move(rhs._path))
+  , _params(std::move(rhs._params))
+  , _completed(rhs._completed)
+  , _terminate_mutex(std::move(rhs._terminate_mutex))
+  , _wait_mutex(std::move(rhs._wait_mutex))
+  , _stdout_consumer(std::exchange(rhs._stdout_consumer, std::nullopt))
+  , _stderr_consumer(std::exchange(rhs._stderr_consumer, std::nullopt))
+  , _gate(std::move(rhs._gate)) {
+    rhs._completed = true;
+}
+
+external_process::~external_process() {
+    vassert(_completed, "Process has not yet completed");
+    vassert(!_stdout_consumer.has_value(), "stdout consumer not cleaned up");
+    vassert(!_stderr_consumer.has_value(), "stderr consumer not cleaned up");
+}
+
+ss::future<ss::experimental::process::wait_status> external_process::wait() {
+    auto guard = _gate.hold();
+    if (_completed) {
+        throw std::system_error(
+          make_error_code(error_code::process_already_completed));
+    }
+
+    auto units = _wait_mutex.try_get_units();
+    if (!units) {
+        throw std::system_error(
+          make_error_code(error_code::process_already_being_waited_on));
+    }
+
+    auto resp = co_await _process.wait();
+
+    if (_stdout_consumer.has_value()) {
+        try {
+            co_await std::move(_stdout_consumer).value();
+        } catch (...) {
+            // ignore the error if `input_stream::consume` throws
```

Comment:
This causes clang-tidy warnings.

Can we instead do:

```
catch (const std::exception& e) {
  std::ignore = e;
}
```

Same for stderr, you could even be fancy and do:


```
for (auto& consumer : {_stdout, _stderr}) {
  if (!consumer) continue;
  try {
    co_await std::move(consumer).value();
  } catch (const std::exception& e) {
    std::ignore = e;
  }
}
```

Reply:
> This causes clang-tidy warnings.

I didn't see any show up.  What do you use to show clang tidy warnings?